automation_tools/satellite6/log.py

```diff
@@ -0,0 +1,130 @@
+
+        :param host: str the hostname
+        """
+        self.host = host
+        self.log_state = {
+            '/var/log/foreman-installer/satellite.log': 0,
+            '/var/log/foreman/production.log': 0,
+            '/var/log/foreman-proxy/proxy.log': 0,
+            '/var/log/candlepin/candlepin.log': 0,
+            '/var/log/messages': 0,
+            '/var/log/mongodb/mongodb.log': 0,
+            '/var/log/tomcat/catalina.out': 0
+        }
+
+    def __enter__(self):
+        """
+        Fetch current line count for Satellite log files
+        :return: LogAnalyzer
+        """
+        self._update_log_files_state()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Analyzes log files checking if some error occurred since last
+        log_state update
+        """
+        print('#### Analyzing logs of host %s' % self.host)
+        self._update_log_files_state()
+
+        def fetch_appended_log_lines():
+            for log_file, lines_appended in self.log_state.items():
+                if lines_appended > 0:
+                    content = run(
+                        'tail -n {} {}'.format(lines_appended, log_file),
+                        quiet=True
+                    )
+                    analyze(log_file, content)
+
+        execute(fetch_appended_log_lines, host=self.host)
+
+    def _update_log_files_state(self):
+        """Update log_dct with adding delta from current number of lines of
+        each item and the last provided by dct.
+
+        So this method can be used to check how many lines were appended on
+        a file during some processes and used to tail them. If log_state is
+        None a new dict is created with initial values been 0 (zero).
+        """
+
+        def get_line_count():
+            for log_file, old_value in self.log_state.items():
+                try:
+                    current_value = int(
+                        run(
+                            'wc -l < {}'.format(log_file),
+                            quiet=True
+                        )
+                    )
+                except ValueError:
+                    self.log_state[log_file] = 0
+                else:
+                    self.log_state[log_file] = current_value - old_value
+
+        execute(get_line_count, host=self.host)
+
+
+def analyze(log_file, content):
+    """Analyzes appended content from a log file. For now it is only
+    checking for ERROR status on regular log files.
+
+    :param log_file: str log file path on host
+    :param content: str with log file content
+    """
+
+    def print_lines(lines_enumeration):
+        """print lines with numbers"""
+        for i, line in lines_enumeration:
+            _print_wrapper('{}: {}'.format(i, line))
+
+    _print_wrapper('### Analyzing %s:' % log_file)
+    _print_wrapper('## Errors found:')
+    content_lines = content.split('\n')
+    error_tokens = (
```

Comment:
I think we should move this somewhere else, the list will increase in future. And its better to have its own place somewhere,